Block/Adminhtml/Insurance/SubscriptionDetails.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Alma\MonthlyPayments\Block\Adminhtml\Insurance;
+
+use Alma\MonthlyPayments\Helpers\ApiConfigHelper;
+use Alma\MonthlyPayments\Helpers\InsuranceHelper;
+use Alma\MonthlyPayments\Helpers\Logger;
+use Alma\MonthlyPayments\Model\Insurance\ResourceModel\Subscription\Collection;
+use Alma\MonthlyPayments\Model\Insurance\ResourceModel\Subscription\CollectionFactory;
+use Magento\Backend\Block\Template;
+use Magento\Directory\Helper\Data as DirectoryHelper;
+use Magento\Framework\Exception\InputException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\Json\Helper\Data as JsonHelper;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Model\OrderRepository;
+
+class SubscriptionDetails extends Template
+{
+    /**
+     * @var InsuranceHelper
+     */
+    private $insuranceHelper;
+    /**
+     * @var ApiConfigHelper
+     */
+    private $apiConfigHelper;
+    /**
+     * @var Logger
+     */
+    private $logger;
+    /**
+     * @var Collection
+     */
+    private $collectionFactory;
+    /**
+     * @var OrderRepository
+     */
+    private $orderRepository;
+
+    public function __construct(
+        Logger            $logger,
+        Template\Context  $context,
+        InsuranceHelper   $insuranceHelper,
+        ApiConfigHelper   $apiConfigHelper,
+        CollectionFactory $collectionFactory,
+        OrderRepository   $orderRepository,
+        array             $data = [],
+        ?JsonHelper       $jsonHelper = null,
+        ?DirectoryHelper  $directoryHelper = null
+    )
+    {
+        parent::__construct(
+            $context,
+            $data,
+            $jsonHelper,
+            $directoryHelper
+        );
+        $this->insuranceHelper = $insuranceHelper;
+        $this->apiConfigHelper = $apiConfigHelper;
+        $this->logger = $logger;
+        $this->collectionFactory = $collectionFactory;
+        $this->orderRepository = $orderRepository;
+    }
+
+    /**
+     * @return string
+     */
+    public function getScriptUrl(): string
+    {
+        return 'https://protect.staging.almapay.com/displayModal.js';
```

Comment:
The url is write on strong

Reply:
Not in this PR ;) error in merge base but fix